### ISSUE TITLE
Rule Request: Prevent moot protocol conformance

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#2750]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Please describe the rule idea, format
this issue's title as `Rule Request: [Rule Name]` and describe:

> 1. Why should this rule be added? Share links to existing discussion about what the community thinks about this.

I sometimes see code use both `Equatable` and `Hashable` protocol conformance, but `Hashable` already inherits from `Equatable`, so the explicit `Equatable` conformance is moot.

> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Would trigger:

```swift
struct Foo: Equatable, RawRepresentable, Hashable {}
```

Would not trigger:

```swift
struct Foo: RawRepresentable, Hashable {}
```

> 3. Should the rule be configurable, if so what parameters should be configurable?

No

> 4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Enabled by default

**Comment by sindresorhus** [comment]
Please keep it open.

**Comment by sindresorhus** [comment]
Please keep it open.
